frontend/app/(auth)/recover-password/__tests__/page.test.tsx

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable max-lines */
+import { fireEvent, render } from "@testing-library/react";
+import { describe, expect, it, vi } from "vitest";
+
+import RecoverPassword from "../page";
+
+const mockUsePathname = vi.fn(() => "/previous-page");
+const mockRedirect = vi.fn((url: string) => ({ url }));
+
+vi.mock("next/navigation", () => ({
+  redirect: (url: string) => mockRedirect(url),
+  usePathname: () => mockUsePathname(),
+}));
+
+const mockUseSupabase = vi.fn(() => ({
+  supabase: {
+    auth: {
+      updateUser: vi.fn(),
+    },
+  },
+  session: {
+    user: {},
+  },
+}));
+
+vi.mock("@/lib/context/SupabaseProvider", () => ({
+  useSupabase: () => mockUseSupabase(),
+}));
+
+const mockPublish = vi.fn();
+
+vi.mock("@/lib/hooks/useToast", () => ({
+  useToast: () => ({
+    publish: mockPublish,
+  }),
+}));
+
+const mockTrack = vi.fn();
+
+vi.mock("@/services/analytics/useEventTracking", () => ({
+  useEventTracking: () => ({
+    track: mockTrack,
+  }),
+}));
+
+describe("RecoverPassword component", () => {
+  it("should render the password update form", () => {
+    mockUseSupabase.mockReturnValue({
+      //@ts-expect-error doing this for testing purposes
+      session: { user: undefined },
+    });
+    const { getByTestId } = render(<RecoverPassword />);
+    const passwordField = getByTestId("password-field");
+    const updateButton = getByTestId("update-button");
+
+    expect(passwordField).toBeDefined();
+    expect(updateButton).toBeDefined();
+  });
+
+  it("should update the password and shows success toast", async () => {
+    const updateUserMock = vi.fn(() => ({
+      data: {},
+    }));
+    mockUseSupabase.mockReturnValue({
+      supabase: {
+        auth: {
+          updateUser: updateUserMock,
+        },
+      },
+      session: { user: {} },
+    });
+    const { getByTestId } = render(<RecoverPassword />);
+    const passwordField = getByTestId("password-field");
+    const updateButton = getByTestId("update-button");
+
+    fireEvent.change(passwordField, { target: { value: "new-password" } });
+    fireEvent.click(updateButton);
+
+    expect(mockTrack).toHaveBeenCalledTimes(1);
+    expect(mockTrack).toHaveBeenCalledWith("UPDATE_PASSWORD");
+
+    return new Promise<void>((resolve) => {
+      setTimeout(() => {
+        expect(updateUserMock).toHaveBeenCalledTimes(1);
```

Comment:
We would like to ensure that `updateUser` has been called with the user's password and not the hardcoded value 'my_super_secret' in case someone changes the code to:

```ts
const { data, error } = await supabase.auth.updateUser({
  password: 'my_super_secret',
});
```

Reply:
Make sense?

Reply:
Good point, see if the next commit would do the job 👍